File: cs50x/pset6/mario/mario.py
from cs50 import get_int

# ask user for height
while True:
    h = get_int("height: ")
    if h > -1 and h < 24:
        break

# print half pyramid


def print_half_pyramid(height):
    for i in range(1, h+1, +1):
        for j in range(h, -1, -1):
            if j > i:
                print(" ", end='')
            else:
                print("#", end='')
        print("")


# call function
print_half_pyramid(h)

File: cs50x/pset1/mario.c
#include <cs50.h>
#include <stdio.h>

// function prototypes
int AskHeight(void);
void PrintHalfPyramid(int h);

int main(void)
{
    // ask user for height of half-pyramid
    int h = AskHeight();
    
    //print half-pyramid
    PrintHalfPyramid(h);
}

/**
* Ask user for height of half-pyramid.
*/

int AskHeight(void)
{
    int h;
    do
    {
        printf("height: ");
        h = GetInt();
    }
    while (h < 0 || h > 23);
    return h;
}

/**
* Prints half-pyramid.
*/

void PrintHalfPyramid(int height) {
    for (int i = 1; i < height + 1; i++)
    {
        for (int j = height; j > 0; j--)
        {
            if (j > i)
            {
                printf(" ");
            } else
            {
                printf("#");
            }
        };
        printf("#\n");
    }
}